android-build-tools/src/ndk.rs

```diff
@@ -0,0 +1,318 @@
+use crate::error::NdkError;
+use crate::target::Target;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Ndk {
+    sdk_path: PathBuf,
+    ndk_path: PathBuf,
+    build_tools_version: String,
+    platforms: Vec<u32>,
+}
+
+impl Ndk {
+    pub fn from_env() -> Result<Self, NdkError> {
+        let sdk_path = {
+            let mut sdk_path = std::env::var("ANDROID_SDK_HOME").ok();
+
+            if sdk_path.is_none() {
+                sdk_path = std::env::var("ANDROID_HOME").ok();
+            }
+
+            PathBuf::from(sdk_path.ok_or(NdkError::SdkNotFound)?)
+        };
+
+        let ndk_path = {
+            let mut ndk_path = std::env::var("ANDROID_NDK_HOME").ok();
+
+            if ndk_path.is_none() {
+                ndk_path = std::env::var("NDK_HOME").ok();
+            }
+
+            // default ndk installation path
+            if ndk_path.is_none() && sdk_path.join("ndk-bundle").exists() {
+                sdk_path.join("ndk-bundle")
+            } else {
+                PathBuf::from(ndk_path.ok_or(NdkError::NdkNotFound)?)
+            }
+        };
+
+        let build_tools_dir = sdk_path.join("build-tools");
+        let build_tools_version = std::fs::read_dir(&build_tools_dir)
+            .or(Err(NdkError::PathNotFound(build_tools_dir)))?
+            .filter_map(|path| path.ok())
+            .filter(|path| path.metadata().unwrap().is_dir())
```

Comment:
To avoid problems with symlinks instead `path.metadata().unwrap().is_dir()` we should do `path.path().is_dir()`.